sql-plugin/src/main/scala/com/nvidia/spark/rapids/lore/replay.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.lore
+
+import ai.rapids.cudf.Table
+import com.nvidia.spark.rapids.{GpuColumnVector, GpuExec}
+import com.nvidia.spark.rapids.Arm.withResource
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.fs.Path
+
+import org.apache.spark.{Partition, SparkContext, TaskContext}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.execution.LeafExecNode
+import org.apache.spark.sql.vectorized.ColumnarBatch
+import org.apache.spark.util.SerializableConfiguration
+
+case class GpuLoreReplayExec(idxInParent: Int, parentRootPath: String) extends LeafExecNode
+  with GpuExec {
+  private lazy val rdd = new GpuLoreReplayRDD(sparkSession.sparkContext,
+    GpuLore.pathOfChild(new Path(parentRootPath), idxInParent).toString)
+  override def output: Seq[Attribute] = rdd.loreRDDMeta.attrs
+
+  override def doExecute(): RDD[InternalRow] = {
+    throw new UnsupportedOperationException("LoreReplayExec does not support row mode")
+  }
+
+  override protected def internalDoExecuteColumnar(): RDD[ColumnarBatch] = {
+    rdd
+  }
+}
+
+class GpuLoreReplayRDD(sc: SparkContext, rootPathStr: String)
+  extends RDD[ColumnarBatch](sc, Nil) with GpuLoreRDD {
+
+  override def rootPath: Path = new Path(rootPathStr)
+
+  private val hadoopConf = sc.broadcast(new SerializableConfiguration(sc.hadoopConfiguration))
```

Comment:
Nit: I think this is going to broadcast a new copy of the Hadoop conf for every GpuLoreReplayRDD.  Seems like the code that initiates the lore loop over the plan should create a single broadcast of the Hadoop conf before the plan traversal and then pass the broadcast to the GpuLoreReplayExec constructor (which then would delegate to GpuLoreReplayRDD) so we reuse the conf rather than create N copies of it.